js/src/utils/Theme.ts

```diff
@@ -15,77 +15,163 @@
+  '--jp-brand-color2': '#90a4ae',
+  '--jp-brand-color3': '#cfd8dc',
+  '--jp-brand-color4': '#eceff1',
+
+  '--jp-accent-color0': '#388e3c',
+  '--jp-accent-color1': '#4caf50',
+  '--jp-accent-color2': '#81c784',
+  '--jp-accent-color3': '#c8e6c9',
+
+  '--jp-warn-color0': '#f57c00',
+  '--jp-warn-color1': '#ff9800',
+  '--jp-warn-color2': '#ffb74d',
+  '--jp-warn-color3': '#ffe0b2',
+
+  '--jp-error-color0': '#d32f2f',
+  '--jp-error-color1': '#f44336',
+  '--jp-error-color2': '#e57373',
+  '--jp-error-color3': '#ffcdd2',
+
+  '--jp-success-color0': '#388e3c',
+  '--jp-success-color1': '#4caf50',
+  '--jp-success-color2': '#81c784',
+  '--jp-success-color3': '#c8e6c9',
+
+  '--jp-info-color0': '#0097a7',
+  '--jp-info-color1': '#00bcd4',
+  '--jp-info-color2': '#4dd0e1',
+  '--jp-info-color3': '#b2ebf2',
+
+  '--md-blue-A100': '#82b1ff',
+  '--md-blue-A200': '#448aff',
+  '--md-blue-A400': '#2979ff',
+  '--md-blue-A700': '#2962ff',
+
+};
+
+export function formatColor(color: string): string {
+  const match = color.match(/rgba\((\d+),\s?(\d+),\s?(\d+),\s?(\d+\.?\d*)\)/);
+  if (match) {
+    const r: number = Math.ceil(Number.parseInt(match[1], 10));
+    const g: number = Math.ceil(Number.parseInt(match[2], 10));
+    const b: number = Math.ceil(Number.parseInt(match[3], 10));
+    const a: number = Math.ceil(Number.parseInt(match[4], 10));
+    const hexColor = CommonUtils.rgbaToHex(r, g, b, a);
+    return hexColor;
+  } else {
+    return color;
+    
   }
+}
 
-  public static getStyle(): DataGrid.Style & { isDark: boolean } {
-    return this.isDark ? this.getDarkStyle() : this.getLightStyle();
+export function evaluateCSSVariable(name: string) {
+  const value = window.getComputedStyle(document.documentElement).getPropertyValue(name);
+  if (value) {
+    return formatColor(value);
+  } else {
+    return defaults[name];
   }
+}
 
-  public static get DEFAULT_DATA_FONT_COLOR(): string {
-    return this.isDark ? '#ffffff' : '#000000';
+export class Theme {
+  public static getStyle(): DataGrid.Style {
+    return {
+      ...DataGrid.defaultStyle,
+      voidColor: evaluateCSSVariable('--jp-layout-color0'),
+      backgroundColor: evaluateCSSVariable('--jp-layout-color1'),
+      rowBackgroundColor: (i) => (i % 2 === 0 ? evaluateCSSVariable('--jp-layout-color1') : evaluateCSSVariable('--jp-layout-color2')),
+      gridLineColor: evaluateCSSVariable('--jp-border-color3'),
+      
+      headerBackgroundColor: evaluateCSSVariable('--jp-layout-color3'),
+      headerGridLineColor: evaluateCSSVariable('--jp-border-color3'),
+    };
   }
 
+  // Header color
   public static get DEFAULT_HEADER_FONT_COLOR(): string {
-    return this.isDark ? '#ffffff' : '#515a5a';
+    return evaluateCSSVariable('--jp-ui-font-color1');
   }
 
-  public static get DEFAULT_HIGHLIGHT_COLOR(): string {
-    return this.isDark ? '#dfdfdf' : '#6ba2c7';
+  public static get DEFAULT_HEADER_BACKGROUND(): string {
+    return evaluateCSSVariable('--jp-layout-color3');
```

Comment:
It would be nice to evaluate those CSS variables only when the Theme changes, and then reuse the evaluation value when needed. Otherwise, we'll end up evaluating them every time it's needed, and this will slow down the table.

Reply:
That's right. Working on this.